### ISSUE TITLE
Tests duplicated in Test Explorer

### ISSUE THREAD

**Issue by oeed** [swellaby/vscode-rust-test-adapter#91]
I'm still fairly new to Rust and Cargo workspace setups, so this is probably a setup issue on my end but I'm not too sure.

Essentially, the tests for all my projects are showing twice:
<img width="342" alt="Screen Shot 2020-09-07 at 17 21 24" src="https://user-images.githubusercontent.com/3835991/92351067-d3a7ca80-f12e-11ea-958e-a684673ce703.png">

In the above example issue-worker is a binary and elasticsearch-wrapper a library. I've setup a workspace with this folder structure. I'll be adding additional libraries in the future and want them all grouped together, hence the lib folder.

```
Cargo.toml [workspace]
issue-worker
  | Cargo.toml [package]
  | src
  | issue-worker.code-workspace
lib
  | elasticsearch-wrapper
      | Cargo.toml [package]
      | src
...
+ other non-Rust folders & files
```

However, a have a multi-root VSCode workspace with the 'issue-worker' and 'lib' folders. The reason for this is this workspace is sitting in a mono-repo that also has other non-Rust projects; I only want to see the relevant Rust folders.

```json
// issue-worker.code-workspace
{
  "folders": [{
      "path": "."
    },
    {
      "path": "../lib"
    }
  ]
}
```

**Comment by calebcartwright** [comment]
Can you share the contents of your workspace Cargo.toml file?

**Comment by oeed** [comment]
Sure

workspace:
```
[workspace]

members = [
  "issue-worker",
  "lib/elasticsearch-wrapper"
]
```

issue-worker:
```
[package]
name = "issue-worker"
version = "0.1.0"
authors = ["..."]
edition = "2018"

# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html

[dependencies]
elasticsearch-wrapper = { path = "../lib/elasticsearch-wrapper" }

```

elasticsearch-wrapper:
```
[package]
name = "elasticsearch-wrapper"
version = "0.1.0"
authors = ["..."]
edition = "2018"

# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html

[dependencies]
elasticsearch = "~7"
serde = "~1"
serde_json = "~1"
```

**Comment by calebcartwright** [comment]
Thanks! Everything looks pretty standard there with the explicit cargo workspace, so that plus the the duplicated `rust` root node in Explorer is highly indicative of the VS Code multiroot workspace doing some double loading somewhere. 

Could you try opening a standalone instance of Code on the cargo workspace root directory?

**Comment by oeed** [comment]
Yep it appears to not duplicate with just the workspace directory open:

Directory:
<img width="292" alt="Screen Shot 2020-09-09 at 13 43 56" src="https://user-images.githubusercontent.com/3835991/92544705-a67c2900-f2a2-11ea-821c-57e7f911aedf.png">

Multi-root Workspace:
<img width="293" alt="Screen Shot 2020-09-09 at 13 44 03" src="https://user-images.githubusercontent.com/3835991/92545117-9153ca00-f2a3-11ea-8551-40a21bc6d103.png">

**Comment by raphaelricardo10** [comment]
I am studying this project to solve this problem. Someone can help me have a overview?

It think the problem is not that hard to solve.

**Comment by calebcartwright** [comment]
That'd be great @raphaelricardo10 thanks!

I don't know off hand what the issue is, but my guess is that the activation is being hit for every workspace:

https://github.com/swellaby/vscode-rust-test-adapter/blob/b0b6b39e375dc90012d20d90d8922e289e10abc2/src/main.ts#L27

The project is obviously very, very light on contributing documentation but it does have the relevant VS Code configurations and npm scripts wired up (debug, testing, etc.) that you should be able to use the VS Code UI directly for such inner dev loop activities

**Comment by mlesin** [comment]
Is there any progress on this?

**Comment by raphaelricardo10** [comment]
Not yet. But there is a workaround: not using multiple VSCode workspaces, but a single workspace with cargo.toml file referring to different rust projects. In my case, I have a root project that consumes different rust libraries, and I added them as git submodules to edit them directly from the consumer.

For example:

`[workspace]

members = [
    "submodules/project-1",
    "submodules/project-2",
    "submodules/project-n",
]`
